backend/app/services/learning-service.py

```
@@ -0,0 +1,110 @@
+
+    def _calculate_similarity_score(self, field_info: Dict[str, Any], learned_mapping: Dict[str, Any]) -> float:
+        field_text = f"{field_info.get('name', '')} {field_info.get('label', '')} {field_info.get('placeholder', '')}".lower()
+        mapping_field = learned_mapping['field_name'].lower()
+
+        field_words = set(field_text.split())
+        mapping_words = set(mapping_field.split())
+
+        if not field_words or not mapping_words:
+            return 0.0
+
+        intersection = field_words.intersection(mapping_words)
```

Ruff (0.12.2)

Local variable `intersection` is assigned to but never used

Remove assignment to unused variable `intersection`

(F841)